google-datacatalog-rdbms-connector/src/google/datacatalog_connectors/rdbms/scrape/metadata_scraper.py

```diff
@@ -77,6 +87,96 @@ def _get_metadata_from_rdbms_connection(self, connection_args, query):
     def _create_dataframe(self, rows):
         return pd.DataFrame(rows)
 
+    def _enrich_metadata_based_on_user_config(self, user_config,
+                                              base_dataframe, connection_args,
+                                              metadata_definition):
+        query_assembler = self._get_query_assembler()
+        enriched_dataframe = base_dataframe
+        if user_config.refresh_metadata_tables:
+            exact_table_names = MetadataNormalizer.\
+                get_exact_table_names_from_dataframe(
+                    base_dataframe, metadata_definition)
+            refresh_queries = query_assembler.get_refresh_metadata_queries(
+                exact_table_names)
+            logging.info('Refreshing metadata')
+            self._refresh_metadata_from_rdbms_connection(
+                connection_args, refresh_queries)
+        if user_config.scrape_optional_metadata:
+            optional_metadata = user_config.get_chosen_metadata_options()
+            optional_queries = query_assembler.get_optional_queries(
+                optional_metadata)
+            logging.info(
+                'Scraping metadata according to configuration file: {}'.format(
+                    optional_metadata))
+            enriched_dataframe = \
+                self._get_optional_metadata_from_rdbms_connection(
+                    connection_args, optional_queries, base_dataframe,
+                    metadata_definition)
+        return enriched_dataframe
+
+    def _refresh_metadata_from_rdbms_connection(self, connection_args,
+                                                refresh_queries):
+        con = None
+        try:
+            con = self._create_rdbms_connection(connection_args)
+            cur = con.cursor()
+            start_update = time.time()
+            for query in refresh_queries:
+                self._execute_refresh_query(cur, query)
+            end_update = time.time()
+            logging.info(
+                'Metadata analysis took {} seconds to run.'
+                'You can turn it off in ingest_cfg.yaml configuration file, '
+                'using refresh_metadata_tables flag'.format(end_update -
+                                                            start_update))
+        except:  # noqa:E722
+            logging.error(
+                'Error connecting to the database to update metadata.')
+            raise
+        finally:
+            if con:
+                con.close()
+
+    def _get_optional_metadata_from_rdbms_connection(self, connection_args,
+                                                     optional_queries,
+                                                     base_dataframe,
+                                                     metadata_definition):
+        con = None
+        try:
+            con = self._create_rdbms_connection(connection_args)
+            cur = con.cursor()
+            for query in optional_queries:
```

Comment:
If you have a list of queries, and one of them throws an error... this seems to be hard to troubleshoot.

Can you log some information about  the query that will be executed at the beginning of the for iteration?